test-framework/arquillian/src/main/java/io/quarkus/arquillian/InjectionEnricher.java

```diff
@@ -0,0 +1,93 @@
+package io.quarkus.arquillian;
+
+import static io.quarkus.arquillian.QuarkusProtocol.convertToTCCL;
+
+import java.lang.reflect.Method;
+
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.spi.BeanManager;
+
+import org.jboss.arquillian.core.api.InstanceProducer;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.test.spi.TestEnricher;
+import org.jboss.arquillian.test.spi.annotation.TestScoped;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Arc;
+
+/**
+ * Enricher that provides method argument injection.
+ */
+public class InjectionEnricher implements TestEnricher {
+    private static final String ANNOTATION_NAME = "javax.inject.Inject";
+
+    private static final Logger log = Logger.getLogger(TestEnricher.class.getName());
+
+    @SuppressWarnings("rawtypes")
+    @Inject
+    @TestScoped
+    private InstanceProducer<CreationalContext> creationalContextProducer;
```

Comment:
What's the purpose of this?

Reply:
It's needed for getting injectable references, I basically took this from https://github.com/arquillian/arquillian-core/blob/1.4.1.Final/testenrichers/cdi/src/main/java/org/jboss/arquillian/testenricher/cdi/CDIInjectionEnricher.java#L103

Reply:
I think Arquillian needs it for this - https://github.com/arquillian/arquillian-core/blob/1.4.1.Final/testenrichers/cdi/src/main/java/org/jboss/arquillian/testenricher/cdi/CreationalContextDestroyer.java#L36
Which looks like automagical CC cleanup :)

Reply:
Does that mean we should have a similar class in Quarkus too? We aren't importing that one

Reply:
Aren't we? In that case why exactly did you need `InstanceProducer`? What consumers were there?
And yes, we should have a place where we invoke `CreationalContext.release()` to properly destroy all dependent instances linked to this `CreationalContext`.

Reply:
I guess you're right, in that case it wasn't quite necessary :) I added the `CreationalContextDestroyer` now